### DISCUSSION TITLE
What's going on with Joi?

### DISCUSSION THREAD

**Discussion by chris48s** [badges/shields#5321] [Ideas]
There's a bunch of re-organisng going on in the hapi js community.

https://github.com/hapijs/hapi/issues/4111
https://github.com/hapijs/hapi/issues/4113
https://github.com/hapijs/hapi/issues/4114

Today is the first time anything has really been said about the future of Joi. It looks like Joi is going to move out of the hapijs organisation and will continue to be maintained, but given our use of Joi in shields is **extensive** it is worth keeping an eye on these developments :eyes:

**Comment by paulmelnikow** [comment]
A new development: sideway/joi#2411

**Reply by chris48s** [reply]
Fun times. Last time we had to do this (in reverse) it wasn't _hard_ as such, but it was _cumbersome_ . Also, we have plugins. Arg

**Comment by chris48s** [comment]
There's an interesting chain of dependencies here.
We depend on joi-extension-semver (which needs to be updated to use joi instead of @hapi/joi in order for us to move), but joi-extension-semver depends on @types/hapi__joi which needs updating before we can do that.
Looks like https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46241 is the first domino that needs to fall so I've subscribed to updates on that PR.

**Comment by chris48s** [comment]
I think now as of version 17.2, Joi now distributes TS types in the package so the fact that https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46241 didn't happen now doesn't matter, which then in-turn unblocks all this :crossed_fingers: 
I'll have a look at it...

**Comment by chris48s** [comment]
maybe this should have been an issue not a discussion. I sort of want to be able to close it now
